src/promptflow-tools/promptflow/tools/azure_content_safety.py

```diff
@@ -0,0 +1,319 @@
+    """
+
+    def __init__(self, connection: AzureContentSafetyConnection):
+        self.connection = connection
+        super(AzureContentSafety, self).__init__()
+
+    @tool
+    def analyze_text(
+            self,
+            text: str,
+            hate_category: TextCategorySensitivity = TextCategorySensitivity.MEDIUM_SENSITIVITY,
+            sexual_category: TextCategorySensitivity = TextCategorySensitivity.MEDIUM_SENSITIVITY,
+            self_harm_category: TextCategorySensitivity = TextCategorySensitivity.MEDIUM_SENSITIVITY,
+            violence_category: TextCategorySensitivity = TextCategorySensitivity.MEDIUM_SENSITIVITY,
+    ):
+        content_safety = ContentSafety(self.connection.endpoint, self.connection.api_key, self.connection.api_version)
+        media_type = MediaType.Text
+        blocklists = []
+
+        detection_result = content_safety.detect(media_type, text, blocklists)
+
+        # Set the reject thresholds for each category
+        reject_thresholds = {
+            Category.Hate: switch_category_threshold(hate_category),
+            Category.SelfHarm: switch_category_threshold(self_harm_category),
+            Category.Sexual: switch_category_threshold(sexual_category),
+            Category.Violence: switch_category_threshold(violence_category),
+        }
+
+        # Make a decision based on the detection result and reject thresholds
+        if self.connection.api_version == "2023-10-01":
+            decision_result = content_safety.make_decision_1001(detection_result, reject_thresholds)
+        else:
+            decision_result = content_safety.make_decision(detection_result, reject_thresholds)
+        return convert_decision_to_json(decision_result)
+
+
+@tool
+def analyze_text(
+        connection: AzureContentSafetyConnection,
+        text: str,
+        hate_category: TextCategorySensitivity = TextCategorySensitivity.MEDIUM_SENSITIVITY,
+        sexual_category: TextCategorySensitivity = TextCategorySensitivity.MEDIUM_SENSITIVITY,
+        self_harm_category: TextCategorySensitivity = TextCategorySensitivity.MEDIUM_SENSITIVITY,
+        violence_category: TextCategorySensitivity = TextCategorySensitivity.MEDIUM_SENSITIVITY,
+):
+    return AzureContentSafety(connection).analyze_text(
+        text=text,
+        hate_category=hate_category,
+        sexual_category=sexual_category,
+        self_harm_category=self_harm_category,
+        violence_category=violence_category,
+    )
+
+
+def switch_category_threshold(sensitivity: TextCategorySensitivity) -> int:
+    switcher = {
+        TextCategorySensitivity.DISABLE: -1,
+        TextCategorySensitivity.LOW_SENSITIVITY: 6,
+        TextCategorySensitivity.MEDIUM_SENSITIVITY: 4,
+        TextCategorySensitivity.HIGH_SENSITIVITY: 2,
+    }
+    return switcher.get(sensitivity, f"Non-supported sensitivity: {sensitivity}")
+
+
+class MediaType(enum.Enum):
```

Comment:
Use `Enum` is enough?

Reply:
This is align with sample code of Azure Content Safety GA version.